### DISCUSSION TITLE
Automatically Hide Popup after Timer

### DISCUSSION THREAD

**Discussion by Bwanna** [ImpulseAdventure/GUIslice#493] [Q&A] [answered]
Hi,
The more I use this library the more interesting functions I discover.  Really want to thank you guys for developing this!

I'm seeking to have a Popup automatically close after a defined period. Here's the code to demonstrate the approach I'm trying. (Please feel free to suggest a different approach!)
This code produces an error for the ENUM reference for the Popup Page.  I've not been able to figure out the correct reference to use... but maybe this isn't the approach?  
`"Compilation error: cannot convert '<unnamed enum>' to 'gslc_tsGui*'"`

BTW, both the FLAG and the pTime=mills(); are set when the Popup Page is initiated.

```
bool FlagState = 0;           //set when Popup Page is shown
unsigned long pTime1,cTime1;
int pTime1_interval = 5000;          //Delay time to close Popup

void setup() {
}

void loop() {  
  cTime1 = millis();
  if (FlagState == 1) {                         //Checks Popup Flag Status
      if (cTime1 - pTime1 > pTime1_interval) {
        gslc_PopupHide(E_PG_POPUP_PAGE1);       //Hide Popup page
      }
    }
}
```

**Comment by Pconti31** [comment, marked as answer]
@Bwanna GUIslice only allows one popup open at a time so there is no need to identify it.
This is the code you want:
```
gslc_PopupHide(&m_gui);
```
Paul--

**Reply by Bwanna** [reply]
Thank you Sir!  Works great.
Cheers,
Steve
